pkg/controller/kubedirectorcluster/members.go

```diff
@@ -1286,3 +1341,65 @@ func fqdnsList(
 	}
 	return strings.Join(fqdns, ",")
 }
+
+// getConnectionVersion will fetch the HashChangeIncrementor from the cluster Annotations
+// if the Annotation isn't available or, for some reason cannot be parsed to an int64,
+// we get the connection version from members that are in "Ready" state
+func getConnectionVersion(
+	reqLogger logr.Logger,
+	cr *kdv1.KubeDirectorCluster,
+	role *roleInfo,
+) int64 {
+
+	if connectionsVersionStr, ok := cr.Annotations[shared.HashChangeIncrementor]; ok {
+		connectionsVersion, connVersionError := strconv.ParseInt(connectionsVersionStr, 10, 64)
+
+		if connVersionError != nil {
+			shared.LogErrorf(
+				reqLogger,
+				connVersionError,
+				cr,
+				shared.EventReasonMember,
+				"Invalid connectionsIncrementor for role{%s}",
+				role.roleStatus.Name,
+			)
+			return getDefaultConnectionVersion(reqLogger, cr, role)
+
+		}
+
+		return connectionsVersion
+
+	}
+
+	return getDefaultConnectionVersion(reqLogger, cr, role)
+
+}
+
+// getDefaultConnectionVersion returns a conneciton version by parsing the version of members in "ready" state
+// if no member is in "Ready" state, we return 0.
+// otherwise, we return the smallest LastConnectionVersion from all the members.
+// This is because we are better off running --reconnect when no connection has changed, rather than not running --reconnect when a connection change does happen
+func getDefaultConnectionVersion(
+	reqLogger logr.Logger,
+	cr *kdv1.KubeDirectorCluster,
+	role *roleInfo,
+) int64 {
+
+	ready := role.membersByState[memberReady]
+
+	if len(ready) == 0 {
+		x := int64(0)
+		return x
+	}
+
+	min := int64(math.MaxInt64)
+	for _, memberStatus := range ready {
+		memberVersion := *memberStatus.StateDetail.LastConnectionVersion
+		if min > memberVersion {
```

Comment:
min is math.MaxInt64, this will be always true?

Reply:
also why max is named min, looks odd

Reply:
We assume a big value(here max int64 value) as min and then assign min if that value > member version. This will get us a minimum member value.

Reply:
if min = math.Max then math.Max will always be  > memberVersion right? Am I missing something  ?